src/test/java/jp/ikedam/jenkins/plugins/extensible_choice_parameter/SystemGroovyChoiceListProviderJenkinsTest.java

```diff
@@ -200,27 +202,35 @@ public void testDescriptor_doFillDefaultChoiceItemsWithoutContext() throws Excep
     public void testDescriptor_doFillDefaultChoiceItemsWithoutPermission() throws Exception
     {
         j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
-        j.jenkins.setAuthorizationStrategy(new GlobalMatrixAuthorizationStrategy());
+        j.jenkins.setAuthorizationStrategy(new MockAuthorizationStrategy()
+                .grant(Jenkins.READ).everywhere().to("user")
+                .grant(Item.READ, Jenkins.READ).everywhere().to("configurer")
+        );
         FreeStyleProject p = j.createFreeStyleProject();
         SystemGroovyChoiceListProvider.DescriptorImpl descriptor = getDescriptor();
-        SecurityContext orig = ACL.impersonate(User.get("user").impersonate());
-        try
-        {
-            descriptor.doFillDefaultChoiceItems(
-                    p,
-                    properScript,
-                    true,
-                    false
-            );
-            fail();
-        }
-        catch(AccessDeniedException e)
-        {
-        }
-        finally
-        {
-            SecurityContextHolder.setContext(orig);
-        }
+
+        WebClient wc = j.createWebClient();
+        wc.getOptions().setThrowExceptionOnFailingStatusCode(false);
+
+        // user has no access to the job => 404
+        User user = User.get("user");
+        wc.login(user.getId());
+
+        // missing Configure permission
+        Page page = wc.getPage(p, descriptor.getDescriptorUrl() +"/fillDefaultChoiceItems/?script=" + properScript + "&sandbox=" + true + "&usePredefinedVariables=" + false);
+        assertEquals(HttpServletResponse.SC_NOT_FOUND, page.getWebResponse().getStatusCode());
+
+        // no job => nothing to reply, duplicate with withoutContext, but using web calls
+        page = wc.goTo(descriptor.getDescriptorUrl() +"/fillDefaultChoiceItems/?script=" + properScript + "&sandbox=" + true + "&usePredefinedVariables=" + false, "application/json");
+        assertEquals(HttpServletResponse.SC_OK, page.getWebResponse().getStatusCode());
+
+        // configurer has access to the job but without Item/Configure permission => 403
+        User configurer = User.get("configurer");
+        wc.login(configurer.getId());
+
+        // missing Configure permission
+        page = wc.getPage(p, descriptor.getDescriptorUrl() +"/fillDefaultChoiceItems/?script=" + properScript + "&sandbox=" + true + "&usePredefinedVariables=" + false);

```

Comment:
JFYI: This test would fail here with `Expected: is "text/html" but: was "application/json"` if the permission check was broken as `getPage()` assumes the response is `HtmlPage`.
It might be better to use `goTo(p.getUrl() + descriptor.getDescriptorUrl() + ..., null)` to focus only to response codes.

Actually, I think the current implementation is OK. Just for information.